vllm/model_executor/models/mamba.py

```diff
@@ -202,17 +201,12 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
 
         self.make_empty_intermediate_tensors = (
             self.backbone.make_empty_intermediate_tensors)
-        if self.scheduler_config is not None and \
-            not self.model_config.enforce_eager:
-            if self.scheduler_config.max_num_seqs > \
-                vllm_config.compilation_config.max_capture_size:
-                self.max_batch_size = \
-                    vllm_config.compilation_config.max_capture_size
-            else:
-                self.max_batch_size = vllm_config.pad_for_cudagraph(
-                    self.scheduler_config.max_num_seqs)
-        else:
-            self.max_batch_size = 8192 + 2
+
+        # Determine max batch size to set size of MambaCache
+        self.max_batch_size = self.scheduler_config.max_num_seqs
+        if not self.model_config.enforce_eager:
+            self.max_batch_size = vllm_config.pad_for_cudagraph(
+                self.max_batch_size)
```

Comment:
Cleaning this up and applying the fix to other Mamba models

`self.scheduler_config` will never be `None`, as noted here: https://github.com/vllm-project/vllm/pull/10996#discussion_r1883022219

When `enforce_eager` is not true, and and the `max_num_seqs  > max_capture_size`, we may not be allocating enough space in the `MambaCache`.

Reply:
nit: wouldn't be bad to pull this out into a function since it seems duplicated, maybe just kept in mamba.py and imported by other models

Reply:
good call, going to throw it in mamba_cache.py

Reply:
turns out we should just push all of this into the MambaCache constructor